### PR TITLE
[VEN-1829]: Deploy Oracles on Sepolia

### PR DESCRIPTION
**Pull request by 0xlucian**
## Description

This PR deploys oracles on Sepolia network and introduces the following changes/additions:
1. Refactor oracle logic for vBNB and native pools support for the oracle
2. Adds deployment config for Sepolia network
3. Adds a VIP based config script, that will print the required VIP targets and calldatas to be executed.

Resolves #VEN-1829

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: helpers/deploymentConfig.ts [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+}
+
+export const addr0000 = "0x0000000000000000000000000000000000000000";
+export const DEFAULT_STALE_PERIOD = 24 * 60 * 60; // 24 hrs
+export const ANY_CONTRACT = ethers.constants.AddressZero;
+
+export const ADDRESSES: PreconfiguredAddresses = {
+  bsctestnet: {
+    vBNBAddress: testnetDeployments.Contracts.vBNB,
+    WBNBAddress: "0xae13d989daC2f0dEbFf460aC112a837C89BAa7cd",
```
Comment (Debugger022) [inline helpers/deploymentConfig.ts:59]:
```suggestion
    WBNBAddress: testnetDeployments.Contracts.WBNB
```

File: deploy/2-configure-feeds.ts [hunk 2]
```diff
@@ -1,334 +1,48 @@
...
+      log: true,
+      deterministicDeployment: false,
+      args: [`Mock${asset.token}`, `Mock${asset.token}`, 18],
+      autoMine: true,
+      contract: "BEP20Harness",
+    });
+
+    const mock = await hre.ethers.getContract(`Mock${asset.token}`);
+
+    console.log(`Configuring resillient oracle for ${asset.token}`);
```
Comment (chechu) [inline deploy/2-configure-feeds.ts:31]:
```suggestion
    console.log(`Configuring resilient oracle for ${asset.token}`);
```

File: deploy/3-vip-based-configuration.ts [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+      commands.push({
+        contract: oraclesData[oracle].underlyingOracle.address,
+        signature: "setMaxStalePeriod(string,uint256)",
+        value: 0,
+        parameters: [tokenConfig],
+      });
+    }
+
+    console.log(``);
+    console.log(`Adding a command to configure resillient oracle for ${asset.token}`);
```
Comment (chechu) [inline deploy/3-vip-based-configuration.ts:75]:
```suggestion
    console.log(`Adding a command to configure resilient oracle for ${asset.token}`);
```

File: helpers/deploymentConfig.ts [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+    VAIAddress: testnetDeployments.Contracts.VAI,
+    pythOracleAddress: "0xd7308b14BF4008e7C7196eC35610B1427C5702EA",
+    sidRegistryAddress: "0xfFB52185b56603e0fd71De9de4F6f902f05EEA23",
+    acm: "0x45f8a08F534f34A97187626E05d4b6648Eeaa9AA",
+    timelock: testnetDeployments.Contracts.Timelock,
+  },
+  bscmainnet: {
+    vBNBAddress: mainnetDeployments.Contracts.vBNB,
+    WBNBAddress: "0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c",
```
Comment (Debugger022) [inline helpers/deploymentConfig.ts:68]:
```suggestion
    WBNBAddress: mainnetDeployments.Contracts.WBNB,
```

File: contracts/ResilientOracle.sol [hunk 5]
```diff
@@ -119,16 +120,19 @@ contract ResilientOracle is PausableUpgradeable, AccessControlledV8, ResilientOr
...
     /// @custom:oz-upgrades-unsafe-allow constructor
     constructor(
-        address vBnbAddress,
+        address nativeMarketAddress,
         address vaiAddress,
         BoundValidatorInterface _boundValidator
-    ) notNullAddress(vBnbAddress) notNullAddress(vaiAddress) notNullAddress(address(_boundValidator)) {
-        vBnb = vBnbAddress;
+    ) notNullAddress(address(_boundValidator)) {
+        nativeMarket = nativeMarketAddress;
```
[the remark this answers is not in the document]
Comment (chechu) [reply contracts/ResilientOracle.sol:135]: I understood the same thing when I saw your changes: those fields will be optional in some networks (at least initially). So, for me it's ok to remove the checks

[the remark this answers is not in the document]
Comment (Debugger022) [reply contracts/ResilientOracle.sol:135]: Got it, I am ok with that.

**Review by chechu — COMMENTED** [review]
After merging #133, we have to add to the `ci.yml` file the following commands in the `export-deployments` task:
```
yarn hardhat export --network sepolia --export ./deployments/sepolia.json
yarn hardhat export --network ethereum --export ./deployments/ethereum.json
```

File: helpers/deploymentConfig.ts [hunk 6]
```diff
@@ -1,40 +1,87 @@
...
+    pythOracleAddress: "0xd7308b14BF4008e7C7196eC35610B1427C5702EA",
+    sidRegistryAddress: "0xfFB52185b56603e0fd71De9de4F6f902f05EEA23",
+    acm: "0x45f8a08F534f34A97187626E05d4b6648Eeaa9AA",
+    timelock: testnetDeployments.Contracts.Timelock,
+  },
+  bscmainnet: {
+    vBNBAddress: mainnetDeployments.Contracts.vBNB,
+    WBNBAddress: mainnetDeployments.Contracts.WBNB,
+    VAIAddress: mainnetDeployments.Contracts.VAI,
+    pythOracleAddress: "0x4D7E825f80bDf85e913E0DD2A2D54927e9dE1594",
```
Comment (chechu) [inline helpers/deploymentConfig.ts:70]: I would add these addresses for `pythOracleAddress`:
* ethereum: `0x4305fb66699c3b2702d4d05cf36551390a4c69c6`
* sepolia: `0x2880ab155794e7179c9ee2e38200202908c17b43`

(According to the [official Pyth documentation](https://docs.pyth.network/documentation/pythnet-price-feeds/evm))

Then I would deploy the PythOracle rerunning the [deploy script](https://github.com/VenusProtocol/oracle/pull/124/files#diff-a6e3e0aedc51ca991268c9403e7a26e9102952a4996186ded649a5478c866b02R110)

File: helpers/deploymentConfig.ts [hunk 7]
```diff
@@ -228,20 +287,57 @@ export const assets: Assets = {
       price: "159990000000000000000",
     },
   ],
+  sepolia: [
+    {
+      token: "WBTC",
+      address: "0x92A2928f5634BEa89A195e7BeCF0f0FEEDAB885b",
+      oracle: "chainlink",
+      price: "25000000000000000000000",
```
Comment (chechu) [inline helpers/deploymentConfig.ts:295]: Are we using the `price` field of `WBTC`, `WETH` and `USDC`. AFAIU we are setting the Chainlink feed in Sepolia for these tokens

**Review by chechu — APPROVED** [review]

**Pull request merged**